### PR TITLE
[RFR] Write changelog and migration guide

### PR DESCRIPTION
**Pull request by jpetitcolas**
First draft. Will be improved over time.

### CODE REVIEW

**Comment by fzaninotto** [comment]
IMO, the migration guide needs an example script with before and after comparison.

**Comment by djhi** [comment]
Not sure if that's possible but I really like the way mui document their changes using diffs:
https://github.com/mui-org/material-ui/releases

**Comment by jpetitcolas** [comment]
I really added the changelog this time.

File: CHANGELOG [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # CHANGELOG
 
+## 4.0.0
+
+* Huge performance boost (almost 10 times faster when zooming or panning)
+* Review configuration to get more intuitive naming
+* Simplify tick format configuration passing only time formats instead of a whole function
+* Fixing zoom and panning center
```
Comment (fzaninotto) [inline CHANGELOG]: s/Fixing/Fix/

File: CHANGELOG [hunk 2]
```diff
@@ -1,5 +1,19 @@
...
+* Better integration with module bundlers (allowing to pass a local D3 object instead of the global one)
+
+We took profit of this major version change to improve the API without taking care of the previous one. See the [migration guide](./MIGRATION-4.0.md) for more informations.
```
Comment (fzaninotto) [inline CHANGELOG]: s/without taking care of the previous one/- unfortunately, we couldn't keep backwards compatibility/

File: MIGRATION-4.0.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Migrating from v3 to v4
+
+## In theory...
+
+API has been deeply reviewed to both simplify configuration and improve overall performances. We didn't take care of backward compatibility during this major changes. Hence, you will probably need to rewrite your current scripts to make it work with this fourth version.
```
Comment (fzaninotto) [inline MIGRATION-4.0.md]: s/API has been deeply reviewed to both simplify configuration and improve overall performances. We didn't take care of backward compatibility during this major changes. Hence, you will probably need to rewrite your current scripts to make it work with this fourth version./The API has been deeply reviewed to both simplify configuration, and improve overall performances. We chose to break backward compatibility to allow the implementation of the new features. That means you will probably need to rewrite your current scripts to make them work with the new version./

File: MIGRATION-4.0.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Here is a mapping of common configuration parameters, before and after this change:
+
+| Former Name       | New Name          | Comments                                                     |
+| ----------------- | ----------------- | ------------------------------------------------------------ |
+| start             | range.start       | Now defaults to one year ago                                 |
```
Comment (fzaninotto) [inline MIGRATION-4.0.md]: please add ticks for better formatting

**Comment by jpetitcolas** [comment]
Back ticks added.

**Pull request merged**